compiler/modules/front.syntax/parseExpr.bal

```diff
@@ -340,6 +340,15 @@ function finishPrimaryExpr(Tokenizer tok, Expr expr, Position startPos) returns
         opPos = tok.currentStartPos();
         check tok.advance();
         Position qnamePos = tok.currentStartPos();
+        if tok.current() == "map" {
```

Comment:
Having a type `SpecialMethodName` as you previously did is good, since these names will grow.